targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/objects.h

```diff
@@ -54,6 +54,14 @@ extern "C" {
 
 #include "nrf_uart.h"
 
+#if DEVICE_CRYPTOCELL
```

Comment:
Should this be `#if defined(DEVICE_CRYPTOCELL)`?

Reply:
mbed-os build system exports some macros with value like ```ABC=1/0```. In that scenario it has to be ```#if ABC``` to work. Though, I am not sure if that is the case here as well.

Reply:
from the compilation, the definition is `-DDEVICE_CRYPTOCELL=1` , so both are valid, because we define `DEVICE_CRYPTOCELL` as 1. I took as example other devices, that are defined this way, such as `#if DEVICE_SERIAL`

Reply:
@RonEld If `#if DEVICE_XXX` is what's commonly used, I'd prefer to use that here. As long as both is valid, that's not a blocker, though.

Reply:
@hanno-arm both are used, and after thinking some more, since I haven't tested on NRF5 boards without CRYPTOCELL, I am not sure what the behavior will be if `DEVICE_CRYPTOCELL is not defined, and we check `#if DEVICE_CRYPTOCELL`

Reply:
Please be aware that it is possible to create a ```mbed_app.json``` in an Mbed OS app where for documentation purpose a user can choose to create a resident macro and only toggle the value ```0/1```. In such scenarios ```#if DEVICE_CRYPTOCELL``` way is better. But if ```DEVICE_CRYPTOCELL``` is defined differently then this does not apply.

>  I am not sure what the behavior will be if ```DEVICE_CRYPTOCELL``` is not defined, and we check ```#if DEVICE_CRYPTOCELL``` .

It would be same as ```#if defined(DEVICE_CRYPTOCELL)```.